src/lib/support/StringSplitter.h

```diff
@@ -0,0 +1,136 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <lib/support/CHIPMemString.h>
+
+#include <stdlib.h>
+#include <string.h>
+
+namespace chip {
+
+/// Provides the ability to split a given string by a character.
+///
+/// Converts things like:
+///   "a,b,c" split by ',': "a", "b", "c"
+///   ",b,c" split by ',': "", "b", "c"
+///   "a,,c" split by ',': "a", "", "c"
+///   "a," split by ',': "a", ""
+///   ",a" split by ',': "", "a"
+///
+///
+/// WARNING: WILL DESTRUCTIVELY MODIFY THE STRING IN PLACE
+///
+class StringSplitter
+{
+public:
+    StringSplitter(char * s, char separator) : mNext(s), mSeparator(separator)
```

Comment:
This cannot work against a `const char *`. Would recommend CharSpans instead:

- Can work with const strings
- No issues with non-null-terminated source strings

Reply:
That is a great suggestion. Applied.